packages/material-ui/src/Container/Container.js

```diff
@@ -1,12 +1,48 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { deepmerge } from '@material-ui/utils';
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import containerClasses, { getContainerUtilityClass } from './containerClasses';
 import capitalize from '../utils/capitalize';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
+const overridesResolver = (props, styles) => {
+  const { fixed, disableGutters, maxWidth = 'lg' } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[`maxWidth${capitalize(String(maxWidth))}`],
+    ...(fixed && styles.fixed),
+    ...(disableGutters && styles.disableGutters),
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes = {}, fixed, disableGutters, maxWidth } = styleProps;
+
+  return {
+    root: clsx(
+      containerClasses.root,
+      classes.root,
+      getContainerUtilityClass(`maxWidth${capitalize(String(maxWidth))}`),
+      {
+        [containerClasses.fixed]: fixed,
+        [containerClasses.disableGutters]: disableGutters,
+      },
+    ),
+  };
+};
```

Comment:
```suggestion
  return {
    root: clsx(
      containerClasses.root,
      classes.root,
      getContainerUtilityClass(`maxWidth${capitalize(String(maxWidth))}`),
      classes[`maxWidth${capitalize(String(maxWidth))}`],
      {
        [containerClasses.fixed]: fixed,
        [classes.fixed]: fixed,
        [containerClasses.disableGutters]: disableGutters,
        [classes.disableGutters]: disableGutters,
      },
    ),
  };
};
```

We need to add both the utility classes + the classes from the `classes` prop. Maybe the name of the method is misleading..